modules/services/getmail.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.getmail;
+
+  retrieverModule = types.submodule ({config,...}: {
+    options = {
+      type = mkOption {
+        type = types.enum [
+          "SimplePOP3Retriever"
+          "SimplePOP3SSLRetriever"
+          "SimpleIMAPRetriever"
+          "SimpleIMAPSSLRetriever"
+        ];
+        default = "SimpleIMAPSSLRetriever";
+        description = "Type of the retriever.";
+      };
+
+      server = mkOption {
+        type = types.string;
```

Comment:
All `types.string` should be replaced by `types.str`, see https://github.com/NixOS/nixpkgs/blob/19ef534fbb044f2b95e0043c6e0f8b582d94952b/lib/types.nix#L203-L205